kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -397,7 +402,28 @@ def _init_and_run_proof(claim_job: KClaimJob) -> tuple[bool, list[str] | None]:
             fallback_on=options.fallback_on,
             interim_simplification=options.interim_simplification,
             no_post_exec_simplify=(not options.post_exec_simplify),
+            port=options.port,
+            haskell_threads=options.max_frontier_parallel,
         ) as kcfg_explore:
+
+            def create_kcfg_explore() -> KCFGExplore:
```

Comment:
When are these resources closed?

Reply:
Would there be anything besides the `KoreClient` that needs closed? Currently that should be being done in `parallel_advance_proof`. When the `with _ProverPool as pool` is exited, the `_ProverPool` calls `close()` on its list of `Prover`s it has created, and the `APRProver` implementation of `close()` closes the `KoreClient` associated with its `KCFGExplore`. However, `parallel_advance_proof` creates a prover on the main thread as well, and I realized that is not getting closed, so I will fix that.

Reply:
Opened a PR for this in pyk here: https://github.com/runtimeverification/k/pull/4323